File: 2.0/csharp/webhooks/createWebhookNonProfitAgency.cs
using System.Collections.Generic;
using CallfireApiClient;
using CallfireApiClient.Api.Common.Model;
using CallfireApiClient.Api.Webhooks.Model;

public class ApiClientSample
{
    public static void Main(string[] args)
    {
        var client = new CallfireClient("api_login", "api_password");
        var webhook = new Webhook
        {
            Name = "Call finished webhook",
            Resource = ResourceType.OUTBOUND_CALL,
            Events = new HashSet<ResourceEvent> {ResourceEvent.FINISHED},
            Callback = "http://callback-service.com/webhook"
        };
        ResourceId resource = client.WebhooksApi.Create(webhook);
    }
}

File: 2.0/java/webhooks/createWebhookNonProfitAgency.java
import com.callfire.api.client.CallfireClient;
import com.callfire.api.client.api.common.model.ResourceId;
import com.callfire.api.client.api.webhooks.model.ResourceType;
import com.callfire.api.client.api.webhooks.model.Webhook;

class ApiClientSample {
    public static void main(String[] args) {
        CallfireClient client = new CallfireClient("api_login", "api_password");
        
        Webhook webhook = new Webhook();
        webhook.setName("OutboundCall webhook");
        webhook.setResource(ResourceType.OUTBOUND_CALL);
        webhook.getEvents().add(ResourceType.ResourceEvent.FINISHED);
        // CallFire will send a POST request to specified endpoint
        webhook.setCallback("https://yoursite.com/callback");
        ResourceId id = client.webhooksApi().create(webhook);
    }
}

